### PR TITLE
clean up deprecation warning suppression

### LINKED ISSUE #794: Clean up places where we use `@deprecated` + `@nowarn` where the latter would suffice
I introduced one such into the codebase yesterday at https://github.com/scala/scala/pull/9785 because I'd forgotten about https://github.com/scala/scala/pull/9707. I can take it out again, but at the same time I should look around to see where else we used this pattern.

### PR DESCRIPTION
**Pull request by SethTisue**
fixes scala/scala-dev#794

CI won't test this on JDK 17 until after we merge it, but I tested it on JDK 17 locally with `set every fatalWarnings := true` followed by `library/compile` and `partest test/files/run/t2318.scala`

### CODE REVIEW

**Comment by som-snytt** [comment]
Makes me wonder why xlint doesn't lint empty strings in `@deprecated("","")`. That is, if you're going to enforce supplying args (which IMHO was over-zealous), you might as well enforce non-empty.

**Review by som-snytt — APPROVED** [review]
how did we get any work done before warning suppression?

offering approval, although lrytz eyes are also required.

File: test/files/run/t2318.scala [hunk 1]
```diff
@@ -4,13 +4,14 @@
...
+import scala.annotation.nowarn
 
-// SecurityManager is deprecated on JDK 17, so we sprinkle `@deprecated` around
+// SecurityManager is deprecated on JDK 17, so we sprinkle `@nowarn` around
 
 object Test {
   trait Bar { def bar: Unit }
 
-  @deprecated
+  @nowarn("cat=deprecation")
```
Comment (som-snytt) [inline test/files/run/t2318.scala:14]: for example, I might have moved deprecation to Test. Although that suppresses the warning everywhere, it's also true that the test itself is deprecated. 

Possibly, the deprecation suppression rule should be scoped: suppress warnings for using deprecated elements in the enclosing scope marked deprecated, not for all deprecated elements.

**Comment by lrytz** [comment]
Thanks!

**Pull request merged**